doc/nginx-reverse-proxy.md

```diff
@@ -0,0 +1,97 @@
+# Use nginx as a reverse proxy
+
+It is possible to deploy StoRM WebDAV using nginx as a reverse proxy.
+
+The main pros of this type of deployment are:
+
+- use nginx to manage TLS termination
+- delegate VOMS proxy authentication to [ngx_http_voms_module](https://baltig.infn.it/cnafsd/ngx_http_voms_module)
+- improve performance of downloads by using nginx to handle GET requests
+
+## How to deploy StoRM WebDAV using nginx
+
+Install nginx and [ngx_http_voms_module](https://baltig.infn.it/cnafsd/ngx_http_voms_module) on your server.
+
+Change the configuration of nginx to:
+
+- enable the client certificates
+- set the correct headers for VOMS authentication
+- add an internal endpoint to which to redirect GET requests
+
+In your `application.yml` configuration set `storm.nginx-reverse-proxy` to `true`.
+
+Example nginx configuration:
+
+```
+server {
+	location /internal-get {
```

Comment:
We need to document that this location must match the prefix used inside the codebase